src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java

```diff
@@ -699,6 +702,58 @@ public List getRepositoryVOList() throws ConstraintException, SystemException, B
 		return repositoryVOList;
     }
 
+	/**
+	 * This method can be used by actions and use-case-controllers that only need to have simple access to the
+	 * functionality. They don't get the transaction-safety but probably just wants to show the info.
+	 */	
+    
+    public List<RepositoryVO> getRepositoryVOList(Database db) throws ConstraintException, SystemException, Bug
+    {   
+		String key = "repositoryVOList";
+		logger.info("key:" + key);
+		List<RepositoryVO> cachedRepositoryVOList = (List<RepositoryVO>)CacheController.getCachedObject("repositoryCache", key);
+		if(cachedRepositoryVOList != null)
+		{
+			logger.info("There was an cached repo list:" + cachedRepositoryVOList.size());
+			return cachedRepositoryVOList;
+		}
+		
+		List<RepositoryVO> repositoryVOList = new ArrayList<RepositoryVO>();
+
+		String SQL = "SELECT r.repositoryId, r.name, r.description, r.dnsName, r.isDeleted from cmRepository r order by r.repositoryId";
+		logger.info("SQL:" + SQL);
+		
+		try
+		{
+			Connection conn = (Connection) db.getJdbcConnection();
+			
+			PreparedStatement psmt = conn.prepareStatement(SQL.toString());
+	
+			ResultSet rs = psmt.executeQuery();
+			while(rs.next())
+			{
+				RepositoryVO repoVO = new RepositoryVO();
+				repoVO.setRepositoryId(new Integer(rs.getString(1)));
+				repoVO.setName(rs.getString(2));
+				repoVO.setDescription(rs.getString(3));
+				repoVO.setDnsName(rs.getString(4));
+				repoVO.setIsDeleted(rs.getBoolean(5));
+				
+				logger.info("Found:" + repoVO);
+				repositoryVOList.add(repoVO);
+			}
+			rs.close();
+			psmt.close();
+		}
+		catch(Exception e)
+		{
+			logger.error("Problem getting repo list: " + e.getMessage(), e);
+		}
+		
+		CacheController.cacheObject("repositoryCache", key, repositoryVOList);
```

Comment:
Should the list be cached even if there was an exception and repositoryVOList is null?